src/main/java/org/terasology/climateConditions/igloo/IglooProvider.java

```diff
@@ -0,0 +1,102 @@
+// Copyright 2020 The Terasology Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+package org.terasology.climateConditions.igloo;
+
+import org.terasology.core.world.CoreBiome;
+import org.terasology.core.world.generator.facets.BiomeFacet;
+import org.terasology.math.TeraMath;
+import org.terasology.math.geom.Rect2i;
+import org.terasology.math.geom.Vector2i;
+import org.terasology.utilities.procedural.Noise;
+import org.terasology.utilities.procedural.WhiteNoise;
+import org.terasology.world.generation.Border3D;
+import org.terasology.world.generation.Facet;
+import org.terasology.world.generation.FacetBorder;
+import org.terasology.world.generation.FacetProviderPlugin;
+import org.terasology.world.generation.GeneratingRegion;
+import org.terasology.world.generation.Produces;
+import org.terasology.world.generation.Requires;
+import org.terasology.world.generation.facets.SeaLevelFacet;
+import org.terasology.world.generation.facets.SurfaceHeightFacet;
+import org.terasology.world.generation.facets.base.BaseFieldFacet2D;
+import org.terasology.world.generator.plugin.RegisterPlugin;
+
+@RegisterPlugin
+@Requires({
+        @Facet(value = SurfaceHeightFacet.class, border = @FacetBorder(sides = Igloo.SIZE * 2)),
+        @Facet(value = BiomeFacet.class, border = @FacetBorder(sides = Igloo.SIZE * 2)),
+        @Facet(value = SeaLevelFacet.class, border = @FacetBorder(sides = Igloo.SIZE * 2))
+})
+@Produces(IglooFacet.class)
+public class IglooProvider implements FacetProviderPlugin {
+    private Noise noise;
+
+    @Override
+    public void setSeed(long seed) {
+        noise = new WhiteNoise(seed + 3);
```

Comment:
Why `+ 3` ?

Reply:
That's just a random number.. that is done mostly to prevent same noise being generated by various different facets and then the positions overlapping that just starts throwing a lot of NPEs... 
it wouldn't make a difference even if the number was say 10 or 9313 or 213...

Reply:
For "magic numbers" like that please create a constant and name it something like `ARBITRARY_OVERLAP_OFFSET` to make it more clear what it does :-)

Reply:
+1 on "magic numbers" - could you please the necessary modifications for this PR?